### ISSUE TITLE
Feature Request: Any plan to implement crazy-cow as an integrated tool ?

### ISSUE THREAD

**Issue by DariusMofakhami** [machinelevel/sp425-crazy-cow#3]
Hi again Eric, an other series of requests/questions ! Again, it's just to share ideas and discuss with you what features are feasible and which are not (or just requiring too much time/skills/changes)

I wonder if it would be possible to integrate the functionality of crazy-cow as a rM tool. Here is how I imagine it : 
You have your remarkable tablet with a keyboard plugged in, and crazy-cow running. 
You clik on a 'typed-text' or 'insert text box' tool on the tool panel, and you got the possibility to draw a rectangular box with your pen, that defines a text box. Then you can just type in your keyboard and see the keys being drawn inside the text box with an automatic carriage return when the text reaches the right edge of the textbox. In an ideal world, you could exit the textbox and edit it later by selecting the tool and selecting back the textbox (this is certainly not compatile with the way your code is working). 
I don't think this is easy/possible to implement, but this would for me be THE achievement (i.e. very close to what rM developpers should have implemented if they were willing to allow text writting in Notebooks. The official feature would ideally come with a large selection of fonts, font sizes, font types, as for a real text editor.)

Maybe something easier to code currently that would actually improve a lot the usability of crazy-cow (in my opinion) would be being able to see or choose where the writting will start. I understand your point with old typewritter machine, but navigating through a whole page with backspace and up/down arrows isn't very effective. I know there is also the alternative to use the selection tool, but again, this reduces the usability (forces you to go back and forth from your keyboard to your pen.)
So may be there could be a way to include a blinking cursor ? (that would for example have the aspect of the selection tool in terms of screen display) I guess the position where the next letter is going to appear is stored somewhere. We could use this info to code the blinking bar with the actual size of the font and using the selection tool aspect. Or if this would imply an unwanted tool change, maybe we could think of a script that would just draw a line with the current selected tool, and 'undo' it just after; and this script would be run only after a short time with no input. So that at least you have a way to know where your cursor is at some point, without needing to write and erase a lot.
And it would be great as well to be able to use left and right key to change the (visible or not) cursor position (in addition to up and down) as currently these arrows causes a page change. (and backspace and space isn't very effective to navigate left and right, especially since you want to make backspace trigger 'undo').

Thanks a lot for your time  if you have some to give in this direction, and thank you any way for the work already done !
Darius

**Comment by machinelevel** [comment]
Yep! These are all good ideas, summarised here:

1. **Activate as rM Tool** - The easiest (and scrappiest) way to do this would be to write an icon directly to the frame buffer, and then try to intercept taps on it. That'd be high-maintenance, and unlikely to work with new rM builds. A better option would be for the rM team to add it. `:D`
2. **Blinking cursor** - ...or some indication of where the insertion point is, but which can be disabled to save battery.
3. **Left/right keys** - Yep, I stayed hands-off on these, as the existing rM code uses them. I also make use of an "undocumented subtle feature" (also known as a "bug"), which is that the left/right rM buttons change the page without changing the Crazy Cow cursor position, allowing me to flip back and check other pages mid-sentence.

While I can't commit to getting these done, I definitely encourage others to (1) use ace programming skills to improve it, or (2) develop ace programming skills, using this very short simple codebase as an exercise. `:]`
